src/main/java/net/openhft/chronicle/network/tcp/FastJ8SocketChannel.java

```diff
@@ -47,7 +47,7 @@ public int read(ByteBuffer buf) throws IOException {
         if (buf == null)
             throw new NullPointerException();
 
-        if (isBlocking() || isClosed() || !IOTools.isDirectBuffer(buf))
+        if (isBlocking() || isClosed() || !IOTools.isDirectBuffer(buf) || !super.isOpen())
```

Comment:
Just means we don't attempt the "fast read" if we know the channel isn't open. The slow read behaves as you'd expect (throwing the ClosedChannelException when the channel is closed).

Reply:
The fast read throws an IOException from some com.sun package about an invalid file handle

Reply:
Probably still a chance we throw the IOException if the disconnect happens between the check and the read, but it's substantially reduced.